test/unit/AssetManagedPair.t.sol

```diff
@@ -141,6 +141,30 @@ contract AssetManagedPairTest is BaseTest
         assertEq(lLiq1, lLiq2);
     }
 
+    function testAdjustManagement_AdjustAfterLoss() external allPairs
+    {
+        // arrange
+        vm.prank(address(_factory));
+        _pair.setManager(_manager);
+
+        _manager.adjustManagement(_pair, 10e18, 10e18);
+        _manager.adjustBalance(_pair, address(_tokenA), 7e18); // 3e18 lost
+
+        // sanity
+        uint256 lTokenAManaged = _manager.getBalance(_pair, address(_tokenA));
+        assertEq(lTokenAManaged, 7e18);
+
+        // act
+        _manager.adjustManagement(_pair, 20e18, 20e18);
+        lTokenAManaged = _manager.getBalance(_pair, address(_tokenA));
+
+        // assert
+        assertEq(lTokenAManaged, 20e18 + 7e18);
+        assertEq(_pair.token0Managed(), 20e18 + 10e18);
+        _pair.sync();
+        assertEq(_pair.token0Managed(), 20e18 + 7e18); // number is updated after sync
+    }
```

Comment:
It would be good to add unit test cases showing that the updated number affects:

- How many shares you get for minting (i.e. you get more)
- How much underlying you get for burning (i.e. you get less)
- How the exchange for the swap changes (i.e. the asset that was lost is more expensive)

This is essentially just saying that "K" adjusts and the pair behaves normally. You can decide if you want to just test K, or write a fuzz test that proves these properties. You could theoretically have two fuzz tests, one showing that any profit results in higher K effects, one that shows any losses results in lower K effects.

Reply:
1. for minting, the mock asset manager doesn't implement the shares system (only the `AaveManager` does) so we can't read and assert the amount of shares
2. for burning, yes we can write a test case to make sure that we indeed get less 
3. for swapping, the the swap does not get cheaper / more expensive because the pair does not sync with the asset manager on swap.

Reply:
I meant:

- Produce a loss
- Call sync()
- Assert `mint` `burn` `swap` on the pair not anywhere else

Reply:
Okay wrote the 3 additional test cases. For `mint` and `burn` they sync with the AM at the start so no need for a `sync`, but swap needs the `sync` to be updated about the loss